src/mono/mono/metadata/icall.c

```diff
@@ -2284,6 +2284,35 @@ typed_reference_to_object (MonoTypedRef *tref, MonoError *error)
 	HANDLE_FUNCTION_RETURN_REF (MonoObject, result);
 }
 
+gpointer
+ves_icall_System_RuntimeFieldHandle_GetFieldDataReference (MonoObjectHandle target, MonoClassField *field, MonoError *error)
+{
+	g_assert (field);
+
+	(void)mono_handle_class (target);
```

Comment:
why?

do you just want a failure if that the object is null? `g_assert (!MONO_HANDLE_IS_NULL(target))`

Reply:
This was copy/paste from the previous implementation. I'm happy to remove it if you think it's okay.

Reply:
Will remove.